pkg/controllerhelpers/scylla.go

```diff
@@ -69,28 +68,33 @@ func GetRequiredScyllaHosts(sc *scyllav1.ScyllaCluster, services map[string]*cor
 	return hosts, nil
 }
 
-func NewScyllaClientFromToken(hosts []string, authToken string) (*scyllaclient.Client, error) {
+func NewScyllaClient(cfg scyllaclient.Config) (*scyllaclient.Client, error) {
 	// TODO: unify logging
 	logger, _ := log.NewProduction(log.Config{
 		Level: zap.NewAtomicLevelAt(zapcore.InfoLevel),
 	})
-
-	cfg := scyllaclient.DefaultConfig(authToken, hosts...)
 	scyllaClient, err := scyllaclient.NewClient(cfg, logger.Named("scylla_client"))
 	if err != nil {
 		return nil, err
 	}
-
 	return scyllaClient, nil
 }
 
-func NewScyllaClientFromSecret(secret *corev1.Secret, hosts []string) (*scyllaclient.Client, error) {
-	token, err := helpers.GetAgentAuthTokenFromSecret(secret)
-	if err != nil {
-		return nil, err
-	}
+func NewScyllaClientFromToken(hosts []string, authToken string) (*scyllaclient.Client, error) {
+	cfg := scyllaclient.DefaultConfig(authToken, hosts...)
+	return NewScyllaClient(cfg)
+}
+
+func NewLocalhostScyllaClient() (*scyllaclient.Client, error) {
+	cfg := scyllaclient.DefaultConfig("", "127.0.0.1")
+	cfg.Scheme = "http"
+	cfg.Port = fmt.Sprintf("%d", naming.ScyllaAPIPort)
```

Comment:
can the port be change in the supplied config?

Reply:
Could you elaborate? I don't get the question